homeassistant/components/lyric/sensor.py

```diff
@@ -0,0 +1,189 @@
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up Lyric sensor based on a config entry."""
+    lyric = hass.data[DOMAIN][DATA_LYRIC_CLIENT]
+
+    try:
+        devices = lyric.devices()
+    except Exception as exception:
+        raise PlatformNotReady from exception
+
+    hass.data[DOMAIN][DATA_LYRIC_DEVICES] = devices
+
+    devices = []
+    for location, device in lyric.devices():
+        if device.indoorTemperature:
+            devices.append(
+                LyricSensor(
+                    device,
+                    location,
+                    hass,
+                    "indoorTemperature",
+                    "Temperature",
+                    hass.config.units.temperature_unit,
+                    "mdi:thermometer",
+                    DEVICE_CLASS_TEMPERATURE,
+                )
+            )
+        if device.indoorHumidity:
+            devices.append(
+                LyricSensor(
+                    device,
+                    location,
+                    hass,
+                    "indoorHumidity",
+                    "Humidity",
+                    "%",
+                    "mdi:water-percent",
+                    DEVICE_CLASS_HUMIDITY,
+                )
+            )
+        if device.outdoorTemperature:
+            devices.append(
+                LyricSensor(
+                    device,
+                    location,
+                    hass,
+                    "outdoorTemperature",
+                    "Temperature Outside",
+                    hass.config.units.temperature_unit,
+                    "mdi:thermometer",
+                    DEVICE_CLASS_TEMPERATURE,
+                )
+            )
+        if device.displayedOutdoorHumidity:
+            devices.append(
+                LyricSensor(
+                    device,
+                    location,
+                    hass,
+                    "displayedOutdoorHumidity",
+                    "Humidity Outside",
+                    "%",
+                    "mdi:water-percent",
+                    DEVICE_CLASS_HUMIDITY,
+                )
+            )
+        if device.nextPeriodTime:
+            devices.append(
+                LyricSensor(
+                    device,
+                    location,
+                    hass,
+                    "nextPeriodTime",
+                    "Next Period Time",
+                    "",
+                    "mdi:clock",
+                    DEVICE_CLASS_TIMESTAMP,
+                )
+            )
+        if device.thermostatSetpointStatus:
+            devices.append(
+                LyricSensor(
+                    device,
+                    location,
+                    hass,
+                    "thermostatSetpointStatus",
+                    "Status",
+                    "",
+                    "mdi:thermostat",
+                    None,
+                )
+            )
+
+    async_add_entities(devices, True)
```

Comment:
Passing `True` means that we update them all before adding them to HA. Is that what you want?